### PR TITLE
Reuse external workflows in Fuzzer CI

### PR DESCRIPTION
**Pull request by hmeriann**
This PR makes `test-fuzzer` reuse `Build DuckDB` and `Fuzzer` workflows and connected to the https://github.com/duckdb/duckdb_sqlsmith/issues/22

### CODE REVIEW

**Comment by hmeriann** [comment]
This is the run where it reuses external workflows https://github.com/duckdb/duckdb_sqlsmith/actions/runs/10578425424

File: .github/workflows/test-fuzzer-ci-still-works.yml [hunk 1]
```diff
@@ -10,88 +10,32 @@ on:
...
+            fuzzer: sqlsmith
+          - enable_verification: true
+            fuzzer: duckfuzz_functions
+    uses: duckdblabs/duckdb-fuzzer-ci/.github/workflows/fuzz_duckdb.yml@main
+    with:
+      fuzzer: ${{ matrix.fuzzer }}
+      data: ${{ matrix.data }}
+      timeout-minutes: 20
+      max_queries: 10
+      dry: false
```
Comment (Tmonster) [inline .github/workflows/test-fuzzer-ci-still-works.yml:40]: dry should be true I think
`dry` is basically meant to mean that the program should be run without attempting to alter any github issues. 
If we want to test that the fuzzer CI still works, we want to test that the fuzzer CI debug build can be built, but we don't need to file any github issues or check other issues

**Review by Tmonster — COMMENTED** [review]
Just one comment, otherwise it looks good

File: .github/workflows/test-fuzzer-ci-still-works.yml (hunk 1 above)
Comment (hmeriann) [reply .github/workflows/test-fuzzer-ci-still-works.yml]: Done

**Review by Tmonster — APPROVED** [review]
thanks!

**Pull request merged**